libsel4test/include/sel4test/test.h

```diff
@@ -98,6 +98,18 @@ typedef struct test_type {
 
 /* Declare a test type.
  * For now, we put the test types in a separate elf section. */
+#if defined(__has_attribute) && __has_attribute(retain)
```

Comment:
Can't we avoid the code duplication and use this instead
```suggestion
#if defined(__has_attribute) && __has_attribute(retain)
#define ATTR_USED_RETAIN __attribute__((used,retain))
#else 
#define ATTR_USED_RETAIN __attribute__((used))
#endif

define DEFINE_TEST_TYPE(.....) \
    USED_RETAIN __attribute__((section("_test_type"))) .....
```